.github/workflows/builds.yml

```diff
@@ -32,25 +32,25 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-20.04, macos-10.15]
-        target: [debug, optim]
-        mpi: [sequential]
+        target: [dbg, opt]
+        mpi: [seq]
```

Comment:
I'm confused by this shouldn't it be either empty or `mpi: [seq, par]`?

Reply:
This can be confusing, I agree.

Here is how that works:

1 - Matrix evaluation. Any field in the matrix becomes a dimension, having a dimension of 1 means that the field will be constant. Github will generate 4 jobs from this matrix. That part is easy.

2 - Include evaluation. Any include that specifies all the fields in the matrix is a new job. This can be used to add a specific scenario. Here, we use include to add a value to `build-system`: make. And we fill the other fields as we want.

3 - Include evaluation (bis). Any include that specifies only a subset of all the parameters is meant to add a parameter to the jobs matching the specified parameters.

Our definition is excluding: cmake + parallel, make + opt + par, make + dbg + seq. 
Why? I can’t really remember. I think we didn’t care much about parallel builds, but thought we would use them to do test the debug target as well.

The situation is borderline: we could as well have a flat description of the 8 jobs.

Reply:
Aha, I see. Thanks for the explanation.